**1. Print the string 'Welcome to CIT590' using three print statements**

In [1]:
print('Welcome ', end='')
print('to ', end='')
print('CIT590', end='')

Welcome to CIT590

**2. Print the result of 5 divided by 2 multiplied by 3, overall result raised to the power 4**

In [2]:
x = 5 / 2 * 3
print(x)
x = x ** 4
print(x)

7.5
3164.0625


**3. Print the result and type of the following:**

1) 1.2/2

2) 4/2

3) 7//2

4) 7.2//2

In [17]:
x = 1.2 / 2
print(x)
type(x)


0.6


float

In [16]:
x = 4 / 2
print(x)
type(x)

2.0


float

In [18]:
x = 7 // 2
print(x)
type(x)

3


int

In [20]:
x = 7.2 // 2
print(x)
type(x)

3.0


float

**4. Print the result of rounding the following numbers:**
1) 3.5

2) 4.5

3) -3.5

4) -4.5

In [22]:
print(round(3.5))
print(round(4.5))
print(round(-3.5))
print(round(-4.5))

4
4
-4
-4


**5. Try and see the result.**

1) Is 3 equal to 3.0?

2) Is 3 equal to 3.01?

3) Is 3 equal to 3.0000000000000001?

In [24]:
print(3 == 3.0)
print(3 == 3.1)
print(3 == 3.000000000001)

True
False
False


**6. Now try these:**

1) print the result of 0.3-0.2

2) print the result of (0.3 - 0.2 == 0.1)

In [1]:
print(0.3 - 0.2)
print(0.3 - 0.2 == 0.1)

0.09999999999999998
False


**7. Run both the following codes:**

1) print('2 + 3 = ' + (2 + 3))

2) print('2 + 3 =', (2 + 3))

Are both these codes trying to do the same thing and run without errors (explain in one line)? If not what is the difference between the two?
Also modify (1) to make it run without any errors.

In [2]:
print('2 + 3 = ' + (2 + 3))

TypeError: can only concatenate str (not "int") to str

In [3]:
print('2 + 3 =', (2 + 3))

2 + 3 = 5


**8. Try converting 1 and 0 to bool values. How about converting "abcd", [], and "" to bool values?**

In [6]:
print(bool(1))
print(bool(0))

True
False


In [7]:
print(bool("abcd"))
print(bool([]))
print(bool(""))

True
False
False


**9. Check whether 'a' is greater than 'C'. And is 'abcd' greater than "acd"?**

In [8]:
print("a" > "C")

print('abcd' > "acd")

True
False


### Next, we will go through exercises involving loops, lists, and functions. Read the comments carefully, some functions must return a result.

**1. Favorite list**

In [34]:
def favorite_list():
    """
    This function prompts (ask for inputs) the user for their top 3 favorite movies, then
    appends each one to a list called "favorite_movies", then returns the list.
    The format for the inputs is up to you, for example, you could ask the user for input three separate times.
    """

    favorite_movies = []
    count = 0
    while count < 3:
        favorite_movies.append(input("please entre 3 favorite movies (%d/3)" %(count+1)))
        count += 1
    # for i in favorite_movies:
    #     if i != "":
    #         print(i)
    return favorite_movies
    # your program goes here!


In [35]:
print(favorite_list())

['1', '2', '3']


**2. Print even numbers**

In [36]:
def print_even_number(lst):
    """
    Loop through the given list of numbers and print the even numbers in the original order
    as they appear in the list.
    For example: [6,3,1,4,2] -> [6,4,2]
    """
    for i in lst:
        if (int(i) % 2):
            lst.remove(i)
    return lst


    # your program goes here!

In [37]:
print_even_number(
    [951, 402, 984, 651, 360, 69, 408, 319, 601, 485, 980, 507, 725, 547, 544, 615, 83, 165, 141, 501, 263, 617,
      865, 575, 219, 390, 984, 592, 236, 105, 942, 941])

[402,
 984,
 360,
 408,
 601,
 980,
 725,
 544,
 83,
 141,
 263,
 865,
 219,
 390,
 984,
 592,
 236,
 942]

**3. Digit sum**

In [38]:
def digit_sum(x):
    """
    This function takes in a positive integer x and returns the sum of
    all the number's digits.
    For example: 257 = 2 + 5 + 7 = 14
    """
    y = str(x)
    digit_sum = 0
    for i in y:
        digit_sum += int(i)
    return digit_sum
    

    # your program goes here!

In [39]:
print(digit_sum(257))

14
